src/app/components/fees-row/components/transaction-fee.tsx

```diff
@@ -0,0 +1,23 @@
+import { Tooltip } from '@stacks/ui';
+import { TransactionSigningSelectors } from '@tests-legacy/page-objects/transaction-signing.selectors';
+
+import { CryptoCurrencies } from '@shared/models/currencies.model';
+import { Money } from '@shared/models/money.model';
+
+import { formatDustUsdAmounts, i18nFormatCurrency } from '@app/common/money/format-money';
+import { Caption } from '@app/components/typography';
+
+interface TransactionFeeProps {
+  fee: string | number;
+  feeCurrencySymbol: CryptoCurrencies;
+  usdAmount: Money | null;
+}
+export function TransactionFee({ fee, feeCurrencySymbol, usdAmount }: TransactionFeeProps) {
+  const feeLabel = (
+    <Caption data-testid={TransactionSigningSelectors.FeeToBePaidLabel}>
+      {fee} {feeCurrencySymbol}
+    </Caption>
+  );
+  if (!usdAmount || usdAmount.amount.isNaN()) return feeLabel;
```

Comment:
Not sure where the `isNaN` method is being pulled from, would it be worth using the static metod instead?

```suggestion
  if (!usdAmount || Number.isNaN(usdAmount.amount)) return feeLabel;
```

Reply:
This is not new code, but I'll look into it.

Reply:
`amount` is a `BigNumber` here from the `Money` type, so the method is from that library.